src/server.c

```diff
@@ -1369,6 +1369,15 @@ void whileBlockedCron() {
 
     latencyEndMonitor(latency);
     latencyAddSampleIfNeeded("while-blocked-cron",latency);
+
+    /* We received a SIGTERM during loading, shutting down here in a safe way,
+     * as it isn't ok doing so inside the signal handler. */
+    if (server.shutdown_asap && server.loading) {
+        serverLog(LL_WARNING,"Received shutdown signal during loading, check the logs for more information");
```

Comment:
i think it may be better to move this log print to the signal handler.
in case something in the loading operation is in an infinite loop, we'll never get here.
also, the `check the logs for more information` is redundant, should probably be  `Received shutdown signal during loading, scheduling shutdown.`